### DISCUSSION TITLE
Atualização Mapos.

### DISCUSSION THREAD

**Discussion by giampaulo** [RamonSilva20/mapos#2187] [Q&A]
Pessoal. Da versão 4.38.0 para a 4.40.0 é seguro atualizar diretamente pela aplicação ?

**Comment by Diego-Vinfotec** [comment]
Boa Noite pessoal, realizei a atualização do MapOS Versão: 3.1.12 para 4.52.0 no Ubuntu 24 LTS 
1- a instalação foi tranquila através  install.sh na raiz do projeto (lembrando que os requisitos já estavam instalados no sistema operacional ex.  PHP = 8.3; MySQL>= 8.0 e Composer >= 2);
2- após finalizar o script tive um erro que a extensão mysqli não estava presente, tive que rodar o comando sudo apt-get install php8.3-mysql no terminal e editar o arquivo mysql.cnf para habilitar a extensão.
3- realizado o dump do banco antigo, tive que editar varias tabelas e incluir varias colunas obrigatórias(na nova versão) para conseguir migrar todos os dados, as principais tabelas foram: usuarios incluir coluna cep; produto incluir coluna codDeBarra; cliente incluir as colunas senha e fornecedor; após as alterações no banco antigo ter inserido as informações necessárias. realizada a restauração no novo banco da versão 4.52.0.
4- os dados foram inseridos mais percebi que as os estavam zeradas nos valores e as vendas também, entendi que as tabelas itens_de_vendas, produto_os e servico_os foi incluida a coluna preco, realizado o acerto no BD todos os valores retornaram.
5- Gostaria de deixar registrado que todas funções e relatório estão funcionando perfeitamente.

Obrigado a todos pelo ótimo projeto!!!
